codescan/schema.go

```diff
@@ -216,7 +216,17 @@ func (s *schemaBuilder) buildFromDecl(_ *entityDecl, schema *spec.Schema) error
 				if ti.IsBuiltin() {
 					break
 				}
+
 				if ti.IsType() {
+					if _, ok := ti.Type.(interface{ TypeParams() *types.TypeParamList }); ok {
+						if underlyingStruct, ok := o.Type().Underlying().(*types.Struct); ok {
+							if err := s.buildFromStruct(s.decl, underlyingStruct, schema, make(map[string]string)); err != nil {
```

Comment:
here we pass the underlying struct that is already generated with proper types, for example, for `type TestResultA GenericResult[string]` the structure will be:

```go
type TestResultA struct {
 	RecordsMatched uint32      `json:"records_matched"`
 	Matches        []string    `json:"matches"`
 }